src/transport_performance/utils/defence.py

```diff
@@ -27,18 +30,20 @@ def _handle_path_like(pth, param_nm):
     Returns
     -------
     pathlib.Path
-        Platform agnostic representation of pth.
+        Platform agnostic representation of pth. On unix-like a PosixPath is
+        returned. On windows a WindowsPath is returned. Both are children of
+        pathlib.Path.
 
     """
-    if not isinstance(pth, (str, pathlib.Path)):
+    if not isinstance(pth, (str, pathlib.Path, pathlib.PosixPath)):
```

Comment:
Am I correct in assuming this addition is redundant? This is because `pathlib.Path` is the parent class of `pathlib.PosixPath`, meaning that any `pathlib.PosixPath` passed is an instance of `pathlib.Path`